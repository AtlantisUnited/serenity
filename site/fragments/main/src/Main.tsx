import React                   from 'react'

import { Box, Column, Layout } from '@ui/layout'
import { Option, Search }      from '@ui/search'
import { Text }                from '@ui/text'

import messages                from './messages'

export const Main = ({
  intl,
  data = [],
  onChangeSearch = () => {},
  onSelectOption = (v) => {},
  onSelectValue = (v) => {},
}) => (
  <Box justifyContent='center' bg='stardust' height='calc(100vh - 56px)'>
    <Layout maxWidth={1200} px={[20, 20, 40]} boxSizing='border-box' width='100%'>
      <Column justifyContent='center'>
        <Layout maxWidth={760}>
          <Text
            color='white'
            fontSize={['massive', 'massive', 'biggest']}
            fontWeight='bold'
            lineHeight='extra'
          >
            {intl.formatMessage(messages.collected)}
          </Text>
        </Layout>
        <Layout flexBasis={[30, 30, 56]} />
        <Layout>
          <Search
            onSearch={onChangeSearch}
            onSelect={onSelectOption}
            onSelectValue={onSelectValue}
            placeholder={intl.formatMessage(messages.looking)}
            size='large'
            optionFilterProp='label'
            autoFocus
          >
            {data.map((item: any) => (
              <Option key={item.id} value={item.id} label={item.name}>
                {item.name}
              </Option>
            ))}
          </Search>
        </Layout>
        <Layout flexBasis={100} />
      </Column>
    </Layout>
  </Box>
)
