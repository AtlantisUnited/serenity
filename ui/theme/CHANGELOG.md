# Change Log

All notable changes to this project will be documented in this file.
See [Conventional Commits](https://conventionalcommits.org) for commit guidelines.

## 0.0.5 (2021-05-10)

**Note:** Version bump only for package @ui/theme

## 0.0.4 (2020-09-28)

### Bug Fixes

- **accounts:** add emotion-server, delete font ([08acf5a](https://github.com/Atlantis-Lab/serenity/commit/08acf5a7169352a606925fdfbbaa18f9917a8106))

## 0.0.3 (2020-06-30)

**Note:** Version bump only for package @ui/theme

## 0.0.2 (2020-06-10)

**Note:** Version bump only for package @ui/theme
