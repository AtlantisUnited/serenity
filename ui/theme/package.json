{
  "name": "@ui/theme",
  "version": "0.0.5",
  "license": "MIT",
  "main": "src/index.ts",
  "devDependencies": {
    "@types/react": "^17.0.6",
    "@types/styled-system": "5.1.4"
  },
  "dependencies": {
    "@emotion/css": "^11.1.3",
    "@emotion/react": "^11.4.0",
    "@emotion/styled": "^11.3.0",
    "react": "^17.0.2",
    "styled-system": "^5.1.0"
  }
}
