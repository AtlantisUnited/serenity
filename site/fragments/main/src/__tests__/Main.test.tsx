import React                     from 'react'
import { IntlProvider, useIntl } from 'react-intl'

import { MockedProvider }        from '@apollo/react-testing'
import { ThemeProvider }         from '@ui/theme'
import { render }                from '@testing-library/react'

import { Main }                  from '../Main'

const TestComponent = () => {
  const intl = useIntl()

  return <Main intl={intl} />
}

describe('Main fragment', () => {
  describe('snapshots', () => {
    it('should match latest render snapshot', () => {
      const { asFragment } = render(
        <MockedProvider mocks={[]} addTypename={false}>
          <IntlProvider locale='ru' defaultLocale='ru' messages={{}}>
            <ThemeProvider>
              <TestComponent />
            </ThemeProvider>
          </IntlProvider>
        </MockedProvider>
      )

      expect(asFragment()).toMatchSnapshot()
    })
  })
})
