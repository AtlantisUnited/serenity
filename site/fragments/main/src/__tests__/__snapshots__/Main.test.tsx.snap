// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main fragment snapshots should match latest render snapshot 1`] = `
<DocumentFragment>
  .emotion-3 {
  font-family: 'Inter';
  font-weight: 600;
  font-size: 12px;
  width: 100%;
  position: relative;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  box-sizing: border-box;
  border-radius: 4px;
  border: 1px solid rgba(0,0,0,0.08);
  background-color: #ffffff;
  outline: none;
}

.emotion-3 ul,
.emotion-3 li {
  margin: 0;
  padding: 0;
  list-style: none;
}

.emotion-3:hover,
.emotion-3.rc-select-focused {
  border: 1px solid rgba(0,0,0,0.2);
}

.emotion-3.rc-select-disabled {
  cursor: not-allowed;
}

.emotion-3.rc-select-disabled .rc-select-selection {
  opacity: 0.3;
  background-color: #dedede;
}

.emotion-3:hover.rc-select-disabled {
  border: 1px solid rgba(0,0,0,0.08);
}

.emotion-3 .rc-select-arrow {
  height: 100%;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0px;
  right: 16px;
}

@media (min-width:40em) {
  .emotion-3 .rc-select-arrow {
    right: 20px;
  }
}

.emotion-3 .rc-select-arrow svg {
  cursor: pointer;
}

.emotion-3 .rc-select-selection {
  outline: none;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  box-sizing: border-box;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.emotion-3 .rc-select-selection .rc-select-selection__rendered {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  width: 100%;
  height: 100%;
}

.emotion-3 .rc-select-selection .rc-select-selection__rendered .rc-select-selection__placeholder {
  position: absolute;
  top: 50%;
  font-weight: 500;
  color: #bebebe;
  cursor: text;
}

.emotion-3 .rc-select-selection .rc-select-selection__rendered ul {
  height: 100%;
  width: 100%;
}

.emotion-3 .rc-select-selection__choice {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: #f3f3f3;
  border-radius: 4px;
  float: left;
  height: 30px;
  box-sizing: border-box;
  padding: 5px 10px 5px 15px;
  margin-right: 5px;
  margin-top: 5px;
  position: relative;
  overflow: hidden;
  cursor: auto;
}

.emotion-3 .rc-select-selection__choice .rc-select-selection__choice__remove {
  margin-left: 10px;
}

.emotion-3 .rc-select-selection__choice .rc-select-selection__choice__remove .rc-select-selection__choice__remove-icon {
  font-style: normal;
  color: #9e9e9e;
  cursor: pointer;
}

.emotion-3 .rc-select-selection__choice .rc-select-selection__choice__remove .rc-select-selection__choice__remove-icon:hover {
  color: #000000;
}

.emotion-3 .rc-select-search {
  width: 100%;
  height: 100%;
  padding: 0;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.emotion-3 .rc-select-search .rc-select-search__field {
  border: none;
  background: transparent;
  outline: 0px;
  width: 100%;
  height: 100%;
  padding: 0px;
  font-weight: 500;
  display: inline-block;
}

.emotion-3 .rc-select-search .rc-select-search__field__wrap {
  height: 100%;
  width: 100%;
  padding-top: 10px;
  padding-bottom: 10px;
  box-sizing: border-box;
}

.emotion-3 .rc-select-search .rc-select-search__field__mirror {
  position: absolute;
  top: -999px;
  left: 0;
  white-space: pre;
}

.emotion-3 .rc-select-selection-selected-value {
  opacity: 1 !important;
  position: absolute;
  top: 50%;
  margin-top: -7.5px;
}

.emotion-4 {
  min-height: 64px;
}

.emotion-4 .rc-select-selection {
  padding-left: 50px;
  padding-right: 50px;
}

@media (min-width:40em) {
  .emotion-4 .rc-select-selection {
    padding-left: 72px;
  }
}

.emotion-4 .rc-select-selection__placeholder {
  font-size: 15px;
  line-height: 1;
  margin-top: -7.5px;
}

.emotion-4 .rc-select-search__field {
  font-size: 15px;
}

.emotion-11 {
  box-sizing: border-box;
  height: calc(100vh - 56px);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  background-color: rgb(158,158,158);
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.emotion-10 {
  max-width: 1200px;
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 20px;
  padding-right: 20px;
  box-sizing: border-box;
}

@media screen and (min-width:40em) {
  .emotion-10 {
    padding-left: 20px;
    padding-right: 20px;
  }
}

@media screen and (min-width:52em) {
  .emotion-10 {
    padding-left: 40px;
    padding-right: 40px;
  }
}

.emotion-9 {
  width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
}

.emotion-1 {
  max-width: 760px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-0 {
  color: rgb(255,255,255);
  font-size: 32px;
  font-weight: 700;
  line-height: 1.3;
  font-family: Inter;
}

@media screen and (min-width:40em) {
  .emotion-0 {
    font-size: 32px;
  }
}

@media screen and (min-width:52em) {
  .emotion-0 {
    font-size: 48px;
  }
}

.emotion-2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 30px;
  -ms-flex-preferred-size: 30px;
  flex-basis: 30px;
}

@media screen and (min-width:40em) {
  .emotion-2 {
    -webkit-flex-basis: 30px;
    -ms-flex-preferred-size: 30px;
    flex-basis: 30px;
  }
}

@media screen and (min-width:52em) {
  .emotion-2 {
    -webkit-flex-basis: 56px;
    -ms-flex-preferred-size: 56px;
    flex-basis: 56px;
  }
}

.emotion-7 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-6 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  position: relative;
  width: 100%;
}

.emotion-5 {
  height: 100%;
  position: absolute;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  top: 0;
  left: 16px;
}

@media (min-width:40em) {
  .emotion-5 {
    left: 24px;
  }
}

.emotion-8 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-basis: 100px;
  -ms-flex-preferred-size: 100px;
  flex-basis: 100px;
}

<div
    class="emotion-11"
    display="flex"
    height="calc(100vh - 56px)"
  >
    <div
      class="emotion-10"
      display="flex"
      width="100%"
    >
      <div
        class="emotion-9"
        display="flex"
        width="100%"
      >
        <div
          class="emotion-1"
          display="flex"
        >
          <span
            class="emotion-0"
            color="white"
            font-family="primary"
            font-size="massive,massive,biggest"
            font-weight="bold"
          >
            Мы собрали лучших мастеров в одном месте
          </span>
        </div>
        <div
          class="emotion-2"
          display="flex"
        />
        <div
          class="emotion-7"
          display="flex"
        >
          <div
            class="emotion-6"
          >
            <div
              class="emotion-3 emotion-4 rc-select rc-select-focused rc-select-combobox rc-select-enabled"
            >
              <div
                aria-autocomplete="list"
                aria-controls="test-uuid"
                aria-expanded="false"
                aria-haspopup="true"
                class="rc-select-selection
            rc-select-selection--single"
                role="combobox"
              >
                <div
                  class="rc-select-selection__rendered"
                >
                  <div
                    class="rc-select-selection__placeholder"
                    style="display: block; user-select: none;"
                    unselectable="on"
                  >
                    Что или кого вы ищете?
                  </div>
                  <ul>
                    <li
                      class="rc-select-search rc-select-search--inline"
                    >
                      <div
                        class="rc-select-search__field__wrap"
                      >
                        <input
                          autocomplete="off"
                          class="rc-select-search__field"
                          value=""
                        />
                        <span
                          class="rc-select-search__field__mirror"
                        >
                           
                        </span>
                      </div>
                    </li>
                  </ul>
                </div>
                <span
                  class="rc-select-arrow"
                  style="user-select: none;"
                  unselectable="on"
                >
                  <svg
                    fill="none"
                    height="24"
                    viewBox="0 0 24 24"
                    width="24"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <mask
                      height="16"
                      id="ArrowForwardIcon"
                      mask-type="alpha"
                      maskUnits="userSpaceOnUse"
                      width="16"
                      x="4"
                      y="4"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M10.5858 5.41425L12 4L20 12L12 20L10.5858 18.5858L16.1837 13H4V11H16.1837L10.5858 5.41425Z"
                        fill="white"
                        fill-rule="evenodd"
                      />
                    </mask>
                    <g
                      mask="url(#ArrowForwardIcon)"
                    >
                      <rect
                        fill="rgb(153, 153, 153)"
                        height="80"
                        width="80"
                        x="-28"
                        y="-28"
                      />
                    </g>
                  </svg>
                </span>
              </div>
            </div>
            <div
              class="emotion-5"
            >
              <svg
                height="24"
                viewBox="0 0 24 24"
                width="24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <g
                  fill="none"
                  fill-rule="evenodd"
                  stroke="none"
                  stroke-width="1"
                >
                  <g
                    fill="#999999"
                    fill-rule="nonzero"
                    transform="translate(-4, -4)"
                  >
                    <path
                      d="M20.6666667,18.6666667 L19.6133333,18.6666667 L19.24,18.3066667 C20.5466667,16.7866667 21.3333333,14.8133333 21.3333333,12.6666667 C21.3333333,7.88 17.4533333,4 12.6666667,4 C7.88,4 4,7.88 4,12.6666667 C4,17.4533333 7.88,21.3333333 12.6666667,21.3333333 C14.8133333,21.3333333 16.7866667,20.5466667 18.3066667,19.24 L18.6666667,19.6133333 L18.6666667,20.6666667 L25.3333333,27.32 L27.32,25.3333333 L20.6666667,18.6666667 Z M12.6666667,18.6666667 C9.34666667,18.6666667 6.66666667,15.9866667 6.66666667,12.6666667 C6.66666667,9.34666667 9.34666667,6.66666667 12.6666667,6.66666667 C15.9866667,6.66666667 18.6666667,9.34666667 18.6666667,12.6666667 C18.6666667,15.9866667 15.9866667,18.6666667 12.6666667,18.6666667 Z"
                    />
                  </g>
                </g>
              </svg>
            </div>
          </div>
        </div>
        <div
          class="emotion-8"
          display="flex"
        />
      </div>
    </div>
  </div>
</DocumentFragment>
`;
